run-tests.sh

```diff
@@ -34,30 +35,38 @@ TESTS_SSP="stacksmash"
 TESTS_ASAN="stacksmash"
 TESTS_UBSAN="ubsan"
 for arch in $ARCHS; do
-    TEST_DIR="$arch"
+  for target_os in $TARGET_OSES; do
+    TEST_DIR="$arch-$target_os"
     mkdir -p $TEST_DIR
     for test in $TESTS_C; do
-        $arch-w64-mingw32-clang $test.c -o $TEST_DIR/$test.exe
+        $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test.exe
     done
     for test in $TESTS_C_DLL; do
-        $arch-w64-mingw32-clang $test.c -shared -o $TEST_DIR/$test.dll -Wl,--out-implib,$TEST_DIR/lib$test.dll.a
+        $arch-w64-$target_os-clang $test.c -shared -o $TEST_DIR/$test.dll -Wl,--out-implib,$TEST_DIR/lib$test.dll.a
     done
     for test in $TESTS_C_LINK_DLL; do
-        $arch-w64-mingw32-clang $test.c -o $TEST_DIR/$test.exe -L$TEST_DIR -l${test%-main}-lib
+        $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test.exe -L$TEST_DIR -l${test%-main}-lib
     done
     TESTS_EXTRA=""
     for test in $TESTS_C_NO_BUILTIN; do
-        $arch-w64-mingw32-clang $test.c -o $TEST_DIR/$test-no-builtin.exe -fno-builtin
+        $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test-no-builtin.exe -fno-builtin
         TESTS_EXTRA="$TESTS_EXTRA $test-no-builtin"
     done
     for test in $TESTS_CPP; do
-        $arch-w64-mingw32-clang++ $test.cpp -o $TEST_DIR/$test.exe
+        if [ "$test" = "tlstest-main" ]; then
+            case $target_os in
+            # DLLs can't be loaded without a Windows package
+            mingw32uwp|mingw32winrt) continue ;;
```

Comment:
Won't this cause the tests to fail, when trying to run all of the `$TESTS_CPP` below at the end? So doesn't this require you to remove `$TESTS_CPP` from the line `for test in $TESTS_C $TESTS_C_LINK_DLL $TESTS_CPP $TESTS_EXTRA $TESTS_SSP; do` and add the actually built ones to `TESTS_EXTRA` like for UBSAN below?

Reply:
Not sure the rest would fail but I split this particular test that requires a `LoadLibrary` and put it in `TESTS_EXTRA` as suggested